packages/solana/src/client.ts

```diff
@@ -110,13 +111,51 @@ export class Web3Modal extends Web3ModalScaffold {
       },
 
       getApprovedCaipNetworksData: async () =>
-        new Promise(resolve => {
-          const result = {
-            approvedCaipNetworkIds: undefined,
-            supportsAllNetworks: true
-          }
+        new Promise(async resolve => {
+          const walletChoice = localStorage.getItem(SolConstantsUtil.WALLET_ID)
+          if (walletChoice?.includes(ConstantsUtil.WALLET_CONNECT_CONNECTOR_ID)) {
+            const provider = await this.WalletConnectConnector.getProvider()
+            if (!provider) {
+              throw new Error(
+                'networkControllerClient:getApprovedCaipNetworks - connector is undefined'
+              )
+            }
+            const ns = provider?.session?.namespaces
+            const nsChains = ns?.['solana']?.chains
+            const result = {
+              supportsAllNetworks: false,
+              approvedCaipNetworkIds: nsChains as CaipNetworkId[] | undefined
+            }
 
-          resolve(result)
+            resolve(result)
+          } else {
+            const provider = SolStoreUtil.state.provider
+            if (provider && provider instanceof StandardWalletAdapter) {
+              const solanaNetworkNameToIdMap: Record<string, string> = {
+                mainnet: '5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp',
```

Comment:
Can be used from `SolanaChainIDs` in `packages/solana/utils/chainPath/constants.ts`?